Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/StraightVacuumSinglePipe.cs

```diff
@@ -0,0 +1,441 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Open Rails.  If not, see <http://www.gnu.org/licenses/>.
+
+
+using Microsoft.Xna.Framework;
+using System;
+using ORTS.Common;
+using Orts.Common;
+using ORTS.Scripting.Api;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Orts.Simulation.RollingStocks.SubSystems.Brakes.MSTS
+{
+    class StraightVacuumSinglePipe : VacuumSinglePipe
+    {
+        public StraightVacuumSinglePipe(TrainCar car)
+            : base(car)
+        {
+
+        }
+
+        bool TrainBrakeIncrease = false;
+        bool TrainBrakeDecrease = false;
+
+        bool BrakePipeIncrease = false;
+        bool BrakePipeDecrease = false;
+
+        public override void Initialize(bool handbrakeOn, float maxVacuumInHg, float fullServVacuumInHg, bool immediateRelease)
+        {
+            CylPressurePSIA = BrakeLine1PressurePSI = Vac.ToPress(fullServVacuumInHg);
+            HandbrakePercent = handbrakeOn & (Car as MSTSWagon).HandBrakePresent ? 100 : 0;
+            VacResPressurePSIA = Vac.ToPress(maxVacuumInHg); // Only used if car coupled to auto braked locomotive
+        }
+
+        public override void InitializeMoving() // used when initial speed > 0
+        {
+
+            BrakeLine1PressurePSI = Vac.ToPress(Car.Train.EqualReservoirPressurePSIorInHg);
+            CylPressurePSIA = Vac.ToPress(Car.Train.EqualReservoirPressurePSIorInHg);
+            VacResPressurePSIA = Vac.ToPress(Car.Train.EqualReservoirPressurePSIorInHg); // Only used if car coupled to auto braked locomotive
+            HandbrakePercent = 0;
+        }
+
+
+        public override void Update(float elapsedClockSeconds)
+        {
+            // Two options are allowed for in this module -
+            // i) Straight Brake operation - lead BP pressure, and brkae cylinder pressure are calculated in this module. BP pressure is reversed compared to vacuum brake, as vacuum creation applies the brake cylinder.
+            // Some functions, such as brake pipe propagation are handled in the vacuum single pipe module, with some functions in the vacuum brake module disabled if the lead locomotive is straight braked.
+            // ii) Vacuum brake operation - some cars could operate as straight braked cars (ie non auto), or as auto depending upon what type of braking system the locomotive had. In this case cars required an auxiliary
+            // reservoir. OR senses this and if a straight braked car is coupled to a auto (vacuum braked) locomotive, and it has an auxilary reservoir fitted then it will use the vacuum single pipe module to manage 
+            // brakes. In this case relevant straight brake functions are disabled in this module.
+
+            MSTSLocomotive lead = (MSTSLocomotive)Car.Train.LeadLocomotive;
```

Comment:
Please note that we prefer to use 'var' whenever possible, especially in brand new code.

Reply:
Can you be a bit more specific, and provide an example?

Reply:
```suggestion
            var lead = (MSTSLocomotive)Car.Train.LeadLocomotive;
```

And

```var vr = NumBrakeCylinders * BrakeCylVolM3 / BrakePipeVolumeM3;```

not

```float vr = NumBrakeCylinders * BrakeCylVolM3 / BrakePipeVolumeM3;```

etc.

Reply:
What is the difference between var and float, and why is it better to use it?

Reply:
Changed as requested

Reply:
`var` allows C# to automatically determine the type (`float`, `int`, `bool`, `MSTSLocomotive`, etc.) of variable using a technique called "type inference." We prefer the `var` keyword because compared to declaring an explicit type, it is briefer and less redundant (observe that you typed `MSTSLocomotive` twice in the current code).

Going back and changing all of your variable declarations would entail a lot of work, so I'm not suggesting you do that--but do try to prefer `var` in future submissions. Note that it is not always possible to do so--such as when you're declaring a variable without an initial value, for example.

Reply:
Ryan,
I agree with you that in `var loco = (MSTSLocomotive)Locomotive` the implicit type is shorter and more readable, as well as for `var text = "someText"` or `var pi = 3.1415f`.

However, for the expression `float vr = NumBrakeCylinders * BrakeCylVolM3 / BrakePipeVolumeM3;` and, in general, when the numeric type being used is not clear from the expression, I would prefer to explicitly set the type of the variable. I think it improves the readability and code analysis, because the explicit declaration ensures that you are using a 'float' and not a 'double' or an 'int'.

What's your opinion about it?

Reply:
To be honest, I agree, and I use explicit typing in my personal code. @twpol told me about our project's preference for `var` in one of my early PR's. Perhaps he could clarify?

Reply:
When James explained this policy to me, I believe he agreed with Cesar:

> in general, when the numeric type being used is not clear from the expression, I would prefer to explicitly set the type of the variable.